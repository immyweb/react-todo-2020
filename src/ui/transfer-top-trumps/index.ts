import { TransferTopTrumps } from './transfer-top-trumps';

export default TransferTopTrumps;
