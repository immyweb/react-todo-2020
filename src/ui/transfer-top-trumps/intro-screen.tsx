import * as React from 'react';

import { Gradient } from './gradient';
import ButtonCta from './button-cta';

import {
  TTLozengeBtn,
  BadgeImageHolder,
  Badge,
  Inner,
  TTHeadline,
  TTSubHeadline,
  IntroHeader,
  IntroInstructions,
  IntroContent,
  IntroInstructionsContent,
  IntroInstructionsContentP,
  BtnHolder,
  Divider,
  BadgeImage,
} from './styles';
import { colours } from '../../styles/colours';

import { IIntro, IClub } from './types';
import { ITheme } from '../../types/theme';

interface IProps extends IIntro {
  className?: string;
  theme: ITheme;
  targetClub: IClub;
  onStartGame(): void;
}

export const IntroScreen: React.SFC<IProps> = ({
  headline,
  subHeadline,
  introHeadline,
  introCopy,
  no,
  yes,
  playBtn,
  theme,
  targetClub,
  onStartGame,
}) => {
  return (
    <div>
      <Inner className={'ttt-intro'}>
        <TTHeadline className={'ttt-intro-headline'} fontColour={theme.primary}>
          {headline}
        </TTHeadline>
        <TTSubHeadline className={'ttt-intro-subheadline'} fontColour={colours.$white}>
          {subHeadline}
        </TTSubHeadline>
        <BadgeImageHolder className={'ttt-intro-badge'}>
          <Gradient colour={targetClub.teamColours} top={30} />
          <Badge size={85}>
            <BadgeImage src={targetClub.badge} alt={targetClub.name} />
          </Badge>
        </BadgeImageHolder>
        <IntroHeader className={'ttt-intro-header'}>{introHeadline}</IntroHeader>
        <IntroContent className={'ttt-intro-content'} color={theme.primary}>
          {introCopy}
        </IntroContent>
        <IntroInstructions className={'ttt-intro-instructions'}>
          <IntroInstructionsContent reverse={true}>
            <ButtonCta
              bkgndTop={colours.$lightRed}
              bkgndBottom={colours.$darkRed}
              iconName={'cross-solid'}
              disabled={true}
            />
            <IntroInstructionsContentP>{no}</IntroInstructionsContentP>
          </IntroInstructionsContent>
          <Divider />
          <IntroInstructionsContent>
            <ButtonCta
              bkgndTop={colours.$lightGreen}
              bkgndBottom={colours.$darkGreen}
              iconName={'tick-solid'}
              disabled={true}
            />
            <IntroInstructionsContentP>{yes}</IntroInstructionsContentP>
          </IntroInstructionsContent>
        </IntroInstructions>
        <BtnHolder className={'ttt-intro-button'}>
          <TTLozengeBtn
            bkgndTop={theme.primary}
            bkgndBottom={theme.secondary}
            onClick={onStartGame}
          >
            {playBtn}
          </TTLozengeBtn>
        </BtnHolder>
      </Inner>
    </div>
  );
};
