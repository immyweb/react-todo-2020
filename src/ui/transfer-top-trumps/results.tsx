import * as React from 'react';

import { ChartHorizontalBar } from './horizontal-bar-chart';
import { Gradient } from './gradient';
import { ResultsProfiles } from './results-profiles';
import { ResultsLabels } from './results-labels';

import {
  TTHeadline,
  TTSubHeadline,
  TTLozengeBtn,
  TTLozengeBtnIcon,
  ResultsInner,
  ResultsCharts,
  BtnHolder,
  BadgeImageHolder,
  Badge,
  Inner,
  BadgeImage,
} from './styles';
import { colours } from '../../styles/colours';

import { IResults, IClub, IPlayer, IVoteCount } from './types';
import { ITheme } from '../../types/theme';

interface IProps extends IResults {
  players: IPlayer[];
  playerVotes: IVoteCount[];
  theme: ITheme;
  targetClub: IClub;
  onGoToLink(): void;
}

export const calculatePercentages = (data: IVoteCount[]): IVoteCount[] => {
  const newData = data.map((item, i) => {
    const totalVotes = item.voteCount + item.noVoteCount;
    return {
      ...item,
      votePercent:
        totalVotes === 0 || item.voteCount === 0
          ? 0
          : Math.round((item.voteCount / totalVotes) * 100),
      noVotePercent:
        totalVotes === 0 || item.voteCount === 0
          ? 0
          : Math.round(100 - (item.voteCount / totalVotes) * 100),
    };
  });

  return newData;
};

export const sortProfiles = (results: IVoteCount[], players: IPlayer[]) => {
  return players.map(player => {
    const matchingResult = results.find(result => {
      return player.id === result.player;
    });
    return matchingResult ? { ...player, voteCount: matchingResult.voteCount } : { ...player };
  });
};

export const Results: React.SFC<IProps> = ({
  headline,
  subHeadline,
  playBtn,
  players,
  theme,
  targetClub,
  onGoToLink,
  playerVotes,
}) => {
  const sortedProfiles = sortProfiles(playerVotes, players);
  const updatedPlayerVotes = calculatePercentages(playerVotes);
  return (
    <div>
      <ResultsInner>
        <TTHeadline className={'ttt-results-headline'} fontColour={theme.primary}>
          {headline}
        </TTHeadline>
        <TTSubHeadline className={'ttt-results-subheadline'} fontColour={colours.$white}>
          {subHeadline}
        </TTSubHeadline>
        <BadgeImageHolder className={'ttt-results-badge'}>
          <Gradient colour={targetClub.teamColours} top={15} height={4} opacity={0.5} />
          <Badge size={50}>
            <BadgeImage src={targetClub.badge} alt={targetClub.name} />
          </Badge>
        </BadgeImageHolder>
        <ResultsCharts className={'ttt-results-chart'}>
          <ResultsLabels sortedProfiles={sortedProfiles} />
          <ResultsProfiles sortedProfiles={sortedProfiles} />
          <ChartHorizontalBar data={updatedPlayerVotes} theme={theme} />
        </ResultsCharts>
      </ResultsInner>
      <Inner className={'ttt-inner'} padding={'0 0 25px'}>
        <BtnHolder className={'ttt-results-button'}>
          <TTLozengeBtn bkgndTop={theme.primary} bkgndBottom={theme.secondary} onClick={onGoToLink}>
            <TTLozengeBtnIcon src={targetClub.badge} alt={targetClub.name} />
            {playBtn}
          </TTLozengeBtn>
        </BtnHolder>
      </Inner>
    </div>
  );
};
