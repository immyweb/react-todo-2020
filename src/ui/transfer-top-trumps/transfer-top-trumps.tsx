import * as React from 'react';

import { transferTinderAPI } from '../../api-services';
import { IntroScreen } from './intro-screen';
import { Card } from './card';
import { Controls } from './controls';
import { ProgressBar } from './progress-bar';
import { Results } from './results';
import Error from '../error';

import { TTContainer, Game } from './styles';

import { IComponentProps } from '../../types/tracking';
import { ITheme } from '../../types/theme';
import { IPlayer, Vote, IClub, IIntro, IResults, IVoteCount, ICard } from './types';

interface IProps extends IComponentProps {
  theme: ITheme;
  introScreen: IIntro;
  cardScreen: ICard;
  resultsScreen: IResults;
  targetClub: IClub;
  players: IPlayer[];
}

interface IState {
  players: IPlayer[];
  currentPlayer: IPlayer;
  showIntro: boolean;
  playGame: boolean;
  showResults: boolean;
  playerVotes: IVoteCount[];
  hasErrored: boolean;
}

interface IVoteIds {
  yes: string[];
  no: string[];
}

export class TransferTopTrumps extends React.Component<IProps, IState> {
  cachedWarmState = {};

  initialState = {
    players: [],
    currentPlayer: {
      id: '',
      fname: '',
      sName: '',
      imageMobile: '',
      imageDesktop: '',
      club: '',
      clubBadge: '',
      age: 0,
      fee: '',
      position: '',
      idx: 0,
    },
    showIntro: true,
    playGame: false,
    showResults: false,
    playerVotes: [],
    hasErrored: false,
  };

  constructor(props: IProps) {
    super(props);
    this.state = this.initialState;

    this.updateCard = this.updateCard.bind(this);
  }

  async componentDidMount() {
    const {
      targetClub: { id: targetClubId },
    } = this.props;

    const newPlayers = this.addPlayerStatus(this.props.players);

    // Get votes from API.
    const playerIds = newPlayers.map((player: IPlayer) => player.id);
    try {
      const playerVotes = await transferTinderAPI.getPlayerVotes(targetClubId, playerIds);

      this.cachedWarmState = {
        playerVotes,
        players: newPlayers,
        currentPlayer: newPlayers[0],
      };

      this.setState(this.cachedWarmState);
    } catch (e) {
      this.setState({ hasErrored: true });
    }
  }

  addPlayerStatus(players: IPlayer[]) {
    return players.map((player, i) => {
      if (i === 0) {
        return {
          ...player,
          voted: false,
          selected: false,
          active: true,
          idx: 0,
        };
      }
      return {
        ...player,
        voted: false,
        selected: false,
        active: false,
        idx: i,
      };
    });
  }

  onStartGame = () => {
    this.setState({
      showIntro: false,
      playGame: true,
    });
  };

  onGoToLink = () => {
    window.location.assign(this.props.resultsScreen.exitUrl);
  };

  updateCurrentPlayer(players: IPlayer[], currentPlayer: IPlayer, vote: Vote) {
    return players.map((player: IPlayer) => {
      if (player.id === currentPlayer.id) {
        return {
          ...player,
          voted: true,
          active: false,
          selected: vote === Vote.no ? false : true,
        };
      }
      return player;
    });
  }

  updateCard(currentPlayer: IPlayer, vote: Vote) {
    const {
      targetClub: { id: targetClubId },
    } = this.props;

    // Update current player info
    const newPlayers = this.updateCurrentPlayer(this.state.players, currentPlayer, vote);

    // Move to next player or complete game if last player
    if (currentPlayer.idx! + 1 < this.state.players.length) {
      const nextPlayer = newPlayers[currentPlayer.idx! + 1];

      const newPlayersUpdated = newPlayers.map((player) => {
        if (player.idx === nextPlayer.idx) {
          return {
            ...player,
            active: true,
          };
        }
        return player;
      });

      this.setState({ currentPlayer: nextPlayer, players: newPlayersUpdated });
    } else if (currentPlayer.idx! + 1 === this.state.players.length) {
      this.finishVoting(newPlayers, targetClubId);
    }
  }

  finishVoting = async (newPlayers: IPlayer[], targetClubId: string) => {
    this.setState({
      playGame: false,
      showResults: true,
      players: newPlayers,
    });

    // Send votes to API.
    const votedPlayers = newPlayers.reduce(
      (currentVotedPlayers: IVoteIds, player: IPlayer) => {
        return player.selected
          ? {
              ...currentVotedPlayers,
              yes: [...currentVotedPlayers.yes, player.id],
            }
          : {
              ...currentVotedPlayers,
              no: [...currentVotedPlayers.no, player.id],
            };
      },
      { yes: [], no: [] },
    );

    await transferTinderAPI.addPlayerVotes(targetClubId, votedPlayers.yes, votedPlayers.no);
  };

  render() {
    const { introScreen, resultsScreen, cardScreen, targetClub, theme } = this.props;
    const { showIntro, playGame, showResults, players, currentPlayer, hasErrored } = this.state;

    return hasErrored ? (
      <Error />
    ) : (
      <TTContainer className="ttt-container">
        {showIntro && (
          <IntroScreen
            className={'ttt-intro-screen'}
            {...introScreen}
            onStartGame={this.onStartGame}
            theme={theme}
            targetClub={targetClub}
          />
        )}
        {playGame && (
          <Game className={'ttt-game'}>
            <Card className={'ttt-card'} {...cardScreen} player={currentPlayer} theme={theme} />
            <Controls className={'ttt-controls'} onCastVote={this.updateCard} currentPlayer={currentPlayer} />
            <ProgressBar players={players} theme={theme} />
          </Game>
        )}
        {showResults && (
          <Results
            {...resultsScreen}
            players={this.state.players}
            playerVotes={this.state.playerVotes}
            theme={theme}
            targetClub={targetClub}
            onGoToLink={this.onGoToLink}
          />
        )}
      </TTContainer>
    );
  }
}
