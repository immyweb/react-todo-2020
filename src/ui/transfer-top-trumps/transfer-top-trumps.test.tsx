import * as React from 'react';
import { shallow } from 'enzyme';

import { transferTinderAPI } from '../../api-services';
import { TransferTopTrumps } from './transfer-top-trumps';
import { podTracker } from '../../utils/pod-tracker-utils';

import { IntroScreen } from './intro-screen';
import { Results } from './results';

import { Game } from './styles';

import { Vote } from './types';

jest.mock('../../api-services', () => ({
  transferTinderAPI: {
    addPlayerVotes: jest.fn(),
  },
}));

jest.mock('../../utils/pod-tracker-utils', () => ({
  podTracker: {
    click: jest.fn(),
  },
}));

const defaultProps = {
  id: 'card',
  theme: {
    primary: 'abc',
    secondary: '123',
  },
  introScreen: {
    headline: '',
    subHeadline: '',
    introHeadline: '',
    introCopy: '',
    no: '',
    yes: '',
    playBtn: '',
  },
  cardScreen: {
    currentClub: '',
    pos: '',
    estFee: '',
    playerAge: '',
  },
  resultsScreen: {
    headline: '',
    subHeadline: '',
    sectionHeading: '',
    playBtn: '',
    exitUrl: '',
  },
  targetClub: {
    id: '',
    name: '',
    badge: '',
    teamColours: '',
  },
  players: [],
};

const blankPlayer = {
  id: '',
  fname: '',
  sName: '',
  imageMobile: '',
  imageDesktop: '',
  club: '',
  clubBadge: '',
  age: 0,
  fee: '',
  position: '',
};

function shallowRender(props?: any) {
  return shallow(
    <TransferTopTrumps
      theme={{ primary: '#000000', secondary: '#ffffff' }}
      introScreen={introScreen}
      cardScreen={cardScreen}
      resultsScreen={resultsScreen}
      targetClub={targetClub}
      players={players}
      {...props}
    />,
  );
}

describe('<TransferTopTrumps />', () => {
  describe('UI', () => {
    it('renders correctly when errored', () => {
      const wrapper = shallowRender();
      expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly when not errored', () => {
      const wrapper = shallowRender();
      wrapper.setState({
        hasErrored: false,
      });
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
    });

    it('sets correct state values on launch', () => {
      const wrapper = shallowRender();
      expect(wrapper.state('showIntro')).toEqual(true);
      expect(wrapper.state('playGame')).toEqual(false);
      expect(wrapper.state('showResults')).toEqual(false);
    });

    it('addPlayerStatus adds the correct status to each player', () => {
      const wrapper = shallowRender();
      const instance = wrapper.instance() as TransferTopTrumps;
      const newPlayers = instance.addPlayerStatus(players);
      expect(newPlayers[0].selected).toEqual(false);
      expect(newPlayers[0].active).toEqual(true);
      expect(newPlayers[0].voted).toEqual(false);
      expect(newPlayers[1].selected).toEqual(false);
      expect(newPlayers[1].active).toEqual(false);
      expect(newPlayers[1].voted).toEqual(false);
      expect(newPlayers[2].selected).toEqual(false);
      expect(newPlayers[2].active).toEqual(false);
      expect(newPlayers[2].voted).toEqual(false);
    });

    it('onStartGame hides <IntroScreen /> and shows <Game />', () => {
      const wrapper = shallowRender();
      const instance = wrapper.instance() as TransferTopTrumps;
      wrapper.setState({
        hasErrored: false,
      });
      instance.onStartGame();
      expect(wrapper.state('showIntro')).toEqual(false);
      expect(wrapper.state('playGame')).toEqual(true);
      expect(wrapper.state('showResults')).toEqual(false);
      wrapper.update();
      const introScreen = wrapper.find(IntroScreen);
      expect(introScreen.length).toEqual(0);
      const gameScreen = wrapper.find(Game);
      expect(gameScreen.length).toEqual(1);
    });

    it('renders <Results /> when showResults is set to true', () => {
      const wrapper = shallowRender();
      wrapper.setState({
        hasErrored: false,
        showIntro: false,
        playGame: false,
        showResults: true,
      });
      const introScreen = wrapper.find(IntroScreen);
      expect(introScreen.length).toEqual(0);
      const ResultsScreen = wrapper.find(Results);
      expect(ResultsScreen.length).toEqual(1);
    });

    it('onGoToLink functions goes to supplied exit url', () => {
      window.location.assign = jest.fn();
      const wrapper = shallowRender();
      const instance = wrapper.instance() as TransferTopTrumps;
      instance.onGoToLink();
      expect(window.location.assign).toBeCalledWith(
        'https://www.thesun.co.uk/sport/football/team/1196656/manchester-united/',
      );
    });

    it('updateCurrentPlayer updates the current player with correct Yes status', () => {
      const currentPlayer = {
        ...blankPlayer,
        id: 'willian',
      };
      const wrapper = shallowRender();
      const instance = wrapper.instance() as TransferTopTrumps;
      const newArr = instance.updateCurrentPlayer(players, currentPlayer, Vote.yes);
      expect(newArr[0].voted).toEqual(true);
      expect(newArr[0].active).toEqual(false);
      expect(newArr[0].selected).toEqual(true);
    });

    it('updateCurrentPlayer updates the current player with correct No status', () => {
      const currentPlayer = {
        ...blankPlayer,
        id: 'willian',
      };
      const wrapper = shallowRender();
      const instance = wrapper.instance() as TransferTopTrumps;
      const newArr = instance.updateCurrentPlayer(players, currentPlayer, Vote.no);
      expect(newArr[0].voted).toEqual(true);
      expect(newArr[0].active).toEqual(false);
      expect(newArr[0].selected).toEqual(false);
    });

    it.skip('updateCard function moves to next player if there are more players left', () => {
      const currentPlayer = {
        ...players[0],
      };
      const wrapper = shallowRender();
      wrapper.setState({
        currentPlayer,
        hasErrored: false,
      });
      wrapper.update();
      const instance = wrapper.instance() as TransferTopTrumps;
      instance.updateCard(currentPlayer, Vote.yes);
      wrapper.update();
      expect(wrapper.state('currentPlayer')).toEqual(players[1]);
    });
  });

  describe('finishVoting', () => {
    it('should update the state, track an event, and send the vote data to the api', () => {
      const wrapper = shallow(<TransferTopTrumps {...defaultProps} />);
      const instance = wrapper.instance() as TransferTopTrumps;
      const newPlayers = [
        {
          ...blankPlayer,
          id: 'good',
          selected: true,
        },
        {
          ...blankPlayer,
          id: 'bad',
          selected: false,
        },
      ];

      instance.finishVoting(newPlayers, 'a heavy stick, used for beating');

      expect(podTracker.click).toHaveBeenCalledWith({
        description: 'Transfer Top Trumps game completed',
        id: 'card',
      });
      expect(transferTinderAPI.addPlayerVotes).toHaveBeenCalledWith(
        'a heavy stick, used for beating',
        ['good'],
        ['bad'],
      );
    });
  });
});

const introScreen = {
  headline: 'Transfer Top Trumps',
  subHeadline: 'Who will you choose?',
  introHeadline: 'How to play',
  introCopy: 'Choose the players you would like your club to sign',
  no: "Tap/Click for 'No'",
  yes: "Tap/Click for 'Yes'",
  playBtn: 'Play now',
};

const cardScreen = {
  currentClub: 'Current Club',
  pos: 'Pos.',
  estFee: 'Est. Fee',
  playerAge: 'Age',
};

const targetClub = {
  id: 'man-utd',
  name: 'Man Utd',
  badge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png',
  teamColours: '#a81507',
};

const resultsScreen = {
  headline: 'Transfer Top Trumps',
  subHeadline: 'The results',
  sectionHeading: 'Leaderboard',
  playBtn: 'Club news',
  exitUrl: 'https://www.thesun.co.uk/sport/football/team/1196656/manchester-united/',
};

const players = [
  {
    id: 'willian',
    fname: '',
    sName: 'Willian',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_desktop_@2x.jpg',
    club: 'Chelsea',
    clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/chelsea_@2x.png',
    age: 29,
    fee: '£60m',
    position: 'CAM',
    idx: 0,
  },
  {
    id: 'a_sandro',
    fname: 'Alex',
    sName: 'Sandro',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_desktop_@2x.jpg',
    club: 'Juventus',
    clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/juventus_@2x1.png',
    age: 27,
    fee: '£60m',
    position: 'LB',
    idx: 1,
  },
  {
    id: 's_m_savic',
    fname: 'Sergej',
    sName: 'Milinković-Savić',
    imageMobile:
      'https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg',
    imageDesktop:
      'https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_desktop_@2x.jpg',
    club: 'SS Lazio',
    clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/lazio_@2x1.png',
    age: 23,
    fee: '£70m',
    position: 'CM',
    idx: 2,
  },
];
