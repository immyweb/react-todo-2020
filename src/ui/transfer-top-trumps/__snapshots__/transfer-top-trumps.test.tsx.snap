// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransferTopTrumps /> UI renders correctly when errored 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TransferTopTrumps
    cardScreen={
      Object {
        "currentClub": "Current Club",
        "estFee": "Est. Fee",
        "playerAge": "Age",
        "pos": "Pos.",
      }
    }
    introScreen={
      Object {
        "headline": "Transfer Top Trumps",
        "introCopy": "Choose the players you would like your club to sign",
        "introHeadline": "How to play",
        "no": "Tap/Click for 'No'",
        "playBtn": "Play now",
        "subHeadline": "Who will you choose?",
        "yes": "Tap/Click for 'Yes'",
      }
    }
    players={
      Array [
        Object {
          "age": 29,
          "club": "Chelsea",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/chelsea_@2x.png",
          "fee": "£60m",
          "fname": "",
          "id": "willian",
          "idx": 0,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg",
          "position": "CAM",
          "sName": "Willian",
        },
        Object {
          "age": 27,
          "club": "Juventus",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/juventus_@2x1.png",
          "fee": "£60m",
          "fname": "Alex",
          "id": "a_sandro",
          "idx": 1,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg",
          "position": "LB",
          "sName": "Sandro",
        },
        Object {
          "age": 23,
          "club": "SS Lazio",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/lazio_@2x1.png",
          "fee": "£70m",
          "fname": "Sergej",
          "id": "s_m_savic",
          "idx": 2,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg",
          "position": "CM",
          "sName": "Milinković-Savić",
        },
      ]
    }
    resultsScreen={
      Object {
        "exitUrl": "https://www.thesun.co.uk/sport/football/team/1196656/manchester-united/",
        "headline": "Transfer Top Trumps",
        "playBtn": "Club news",
        "sectionHeading": "Leaderboard",
        "subHeadline": "The results",
      }
    }
    targetClub={
      Object {
        "badge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png",
        "id": "man-utd",
        "name": "Man Utd",
        "teamColours": "#a81507",
      }
    }
    theme={
      Object {
        "primary": "#000000",
        "secondary": "#ffffff",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backgroundColour": "#ffffff",
      "theme": Object {
        "primary": "#eb1701",
        "secondary": "#a5171c",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backgroundColour": "#ffffff",
        "theme": Object {
          "primary": "#eb1701",
          "secondary": "#a5171c",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<TransferTopTrumps /> UI renders correctly when not errored 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TransferTopTrumps
    cardScreen={
      Object {
        "currentClub": "Current Club",
        "estFee": "Est. Fee",
        "playerAge": "Age",
        "pos": "Pos.",
      }
    }
    introScreen={
      Object {
        "headline": "Transfer Top Trumps",
        "introCopy": "Choose the players you would like your club to sign",
        "introHeadline": "How to play",
        "no": "Tap/Click for 'No'",
        "playBtn": "Play now",
        "subHeadline": "Who will you choose?",
        "yes": "Tap/Click for 'Yes'",
      }
    }
    players={
      Array [
        Object {
          "age": 29,
          "club": "Chelsea",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/chelsea_@2x.png",
          "fee": "£60m",
          "fname": "",
          "id": "willian",
          "idx": 0,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg",
          "position": "CAM",
          "sName": "Willian",
        },
        Object {
          "age": 27,
          "club": "Juventus",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/juventus_@2x1.png",
          "fee": "£60m",
          "fname": "Alex",
          "id": "a_sandro",
          "idx": 1,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/SANDRO_mobile_@2x.jpg",
          "position": "LB",
          "sName": "Sandro",
        },
        Object {
          "age": 23,
          "club": "SS Lazio",
          "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/lazio_@2x1.png",
          "fee": "£70m",
          "fname": "Sergej",
          "id": "s_m_savic",
          "idx": 2,
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_desktop_@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/MILINKOVIC-SAVIC_mobile_@2x.jpg",
          "position": "CM",
          "sName": "Milinković-Savić",
        },
      ]
    }
    resultsScreen={
      Object {
        "exitUrl": "https://www.thesun.co.uk/sport/football/team/1196656/manchester-united/",
        "headline": "Transfer Top Trumps",
        "playBtn": "Club news",
        "sectionHeading": "Leaderboard",
        "subHeadline": "The results",
      }
    }
    targetClub={
      Object {
        "badge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png",
        "id": "man-utd",
        "name": "Man Utd",
        "teamColours": "#a81507",
      }
    }
    theme={
      Object {
        "primary": "#000000",
        "secondary": "#ffffff",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          className="ttt-intro-screen"
          headline="Transfer Top Trumps"
          introCopy="Choose the players you would like your club to sign"
          introHeadline="How to play"
          no="Tap/Click for 'No'"
          onStartGame={[Function]}
          playBtn="Play now"
          subHeadline="Who will you choose?"
          targetClub={
            Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png",
              "id": "man-utd",
              "name": "Man Utd",
              "teamColours": "#a81507",
            }
          }
          theme={
            Object {
              "primary": "#000000",
              "secondary": "#ffffff",
            }
          }
          yes="Tap/Click for 'Yes'"
        />,
        false,
        false,
      ],
      "className": "ttt-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "ttt-intro-screen",
          "headline": "Transfer Top Trumps",
          "introCopy": "Choose the players you would like your club to sign",
          "introHeadline": "How to play",
          "no": "Tap/Click for 'No'",
          "onStartGame": [Function],
          "playBtn": "Play now",
          "subHeadline": "Who will you choose?",
          "targetClub": Object {
            "badge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png",
            "id": "man-utd",
            "name": "Man Utd",
            "teamColours": "#a81507",
          },
          "theme": Object {
            "primary": "#000000",
            "secondary": "#ffffff",
          },
          "yes": "Tap/Click for 'Yes'",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            className="ttt-intro-screen"
            headline="Transfer Top Trumps"
            introCopy="Choose the players you would like your club to sign"
            introHeadline="How to play"
            no="Tap/Click for 'No'"
            onStartGame={[Function]}
            playBtn="Play now"
            subHeadline="Who will you choose?"
            targetClub={
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png",
                "id": "man-utd",
                "name": "Man Utd",
                "teamColours": "#a81507",
              }
            }
            theme={
              Object {
                "primary": "#000000",
                "secondary": "#ffffff",
              }
            }
            yes="Tap/Click for 'Yes'"
          />,
          false,
          false,
        ],
        "className": "ttt-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "ttt-intro-screen",
            "headline": "Transfer Top Trumps",
            "introCopy": "Choose the players you would like your club to sign",
            "introHeadline": "How to play",
            "no": "Tap/Click for 'No'",
            "onStartGame": [Function],
            "playBtn": "Play now",
            "subHeadline": "Who will you choose?",
            "targetClub": Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/manutd_@2x.png",
              "id": "man-utd",
              "name": "Man Utd",
              "teamColours": "#a81507",
            },
            "theme": Object {
              "primary": "#000000",
              "secondary": "#ffffff",
            },
            "yes": "Tap/Click for 'Yes'",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
